AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -126,14 +126,17 @@ - (AVPlayerItem *)constructPlayerItem
 {
   ASDN::MutexLocker l(__instanceLock__);
 
-  if (_asset != nil) {
-    AVPlayerItem *playerItem = [[AVPlayerItem alloc] initWithAsset:_asset];
-    playerItem.videoComposition = _videoComposition;
-    playerItem.audioMix = _audioMix;
-    return playerItem;
+  AVPlayerItem *playerItem = nil;
+  if (self.assetURL != nil) {
```

Comment:
In this case you described, the method won't be invoked because this method is invoked in 3 cases:

1) fetch data is called from parent node 
2) fetch data is called from setting the AssetURL
3) fetch data is called from setting the asset. 

In #1, It shouldn't load the video asset more than once. 
In #2, if we already have an _asset, which was created from this assetURL, the setter will be returned without setting the Asset and invoking the player item construction. 
In #3, It's very unlikely that the application will set the assetURL first and then set the Asset.  @maicki and I have discussed how we can prevent users from setting both the asset and url, and @maicki suggested we should add clear documentation for our users.